core/dive.c

```diff
@@ -1328,8 +1329,24 @@ static void fixup_surface_pressure(struct dive *dive)
 			nr++;
 		}
 	}
-	if (nr)
-		dive->surface_pressure.mbar = (sum + nr / 2) / nr;
+	res.mbar = nr ? (sum + nr / 2) / nr : 0;
+	return res;
+}
+
+static void fixup_surface_pressure(struct dive *dive)
+{
+	dive->surface_pressure = calculate_surface_pressure(dive);
+}
+
+/* if the surface pressure in the dive data is redundant to the calculated
+ * value (i.e., it was added by running fixup on the dive) return 0,
+ * otherwise return the air temperature given in the dive */
```

Comment:
the comment should say `surface pressure` instead of `air temperature`, correct?